### PR TITLE
Release/Rails 5.2 Upgrade

### PR DESCRIPTION
**Pull request by peggles2**
Upgrade rails to version 5.2

### CODE REVIEW

File: db/structure.sql [hunk 1]
```diff
@@ -1930,6 +1930,7 @@ INSERT INTO `schema_migrations` (version) VALUES
 ('20181213153332'),
 ('20190205200912'),
 ('20190920181828'),
-('20191113214448');
+('20191113214448'),
+('20191114215421');
```
Comment (MothOnMars) [inline db/structure.sql]: What is the additional migration id? There are no db changes in this PR, so there shouldn't be any changes in `structure.sql`.

File: features/blended_search.feature [hunk 2]
```diff
@@ -1,4 +1,4 @@
-Feature: Blended Search
+  Feature: Blended Search
```
Comment (MothOnMars) [inline features/blended_search.feature]: Remove the extra spaces that were added before `Feature: Blended Search`

**Comment by MothOnMars** [comment]
I don't see `config/initializers/new_framework_defaults_5_2.rb` or any other config changes - was `app:update` run?

File: config/environments/development.rb [hunk 3]
```diff
@@ -46,17 +53,9 @@
   config.assets.quiet = true
 
   # Raises error for missing translations
-  config.action_view.raise_on_missing_translations = true
+  # config.action_view.raise_on_missing_translations = true
```
Comment (MothOnMars) [inline config/environments/development.rb]: uncomment this line - we want those errors to be raised

File: config/environments/development.rb [hunk 4]
```diff
@@ -46,17 +53,9 @@
...
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
 end
-
-ADDITIONAL_BING_PARAMS = { 'traffictype' => 'test' }
-
-DEFAULT_CACHE_DURATION = 6.hours
-AZURE_CACHE_DURATION = 1.day
-BING_CACHE_DURATION = 1.day
-GOOGLE_CACHE_DURATION = 5.minutes
-I14Y_CACHE_DURATION = 1.day
```
Comment (MothOnMars) [inline config/environments/development.rb]: Restore these lines - those constants are used in the app

File: config/initializers/content_security_policy.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+#   # policy.report_uri "/csp-violation-report-endpoint"
+# end
+
+# If you are using UJS then enable automatic nonce generation
+# Rails.application.config.content_security_policy_nonce_generator = -> request { SecureRandom.base64(16) }
+
+# Report CSP violations to a specified URI
+# For further information see the following documentation:
+# https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy-Report-Only
+# Rails.application.config.content_security_policy_report_only = true
```
Comment (MothOnMars) [inline config/initializers/content_security_policy.rb]: We should have a follow-up JIRA ticket to investigate/test/implement these settings. This seems like a very useful set of configs to secure our app: https://guides.rubyonrails.org/security.html#content-security-policy

File: config/initializers/new_framework_defaults_5_2.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+# Store boolean values are in sqlite3 databases as 1 and 0 instead of 't' and
+# 'f' after migrating old data.
+# Rails.application.config.active_record.sqlite3.represent_boolean_as_integer = true
+
+# Use SHA-1 instead of MD5 to generate non-sensitive digests, such as the ETag header.
+# Rails.application.config.active_support.use_sha1_digests = true
+
+# Make `form_with` generate id attributes for any generated HTML tags.
+# Rails.application.config.action_view.form_with_generates_ids = true
```
Comment (MothOnMars) [inline config/initializers/new_framework_defaults_5_2.rb]: We'll need a follow-up ticket to flip all of these flags to the new default.

File: config/environments/development.rb (hunk 4 above)
Comment (peggles2) [reply config/environments/development.rb]: wow I thought I told it to ignore that. will add that back.

File: config/initializers/content_security_policy.rb (hunk 5 above)
Comment (peggles2) [reply config/initializers/content_security_policy.rb]: Sure go ahead create a Jira ticket. We probably have to prioritize that one.

File: config/initializers/new_framework_defaults_5_2.rb (hunk 6 above)
Comment (peggles2) [reply config/initializers/new_framework_defaults_5_2.rb]: srch-1058 would handle that.

Comment (MothOnMars) [reply config/initializers/new_framework_defaults_5_2.rb]: OK - I'll add these to the done criteria for that story

File: config/initializers/content_security_policy.rb (hunk 5 above)
Comment (MothOnMars) [reply config/initializers/content_security_policy.rb]: https://cm-jira.usa.gov/browse/SRCH-1166

**Pull request merged**